bigbluebutton-html5/imports/ui/components/common/meeting-remaining-time-graphql/component.tsx

```diff
@@ -0,0 +1,253 @@
+    fromBreakoutPanel,
+    displayAlerts,
+    isBreakout,
+  } = props;
+
+  const intl = useIntl();
+  const [timeSync] = useTimeSync();
+  const timeRemainingInterval = React.useRef<ReturnType<typeof setTimeout>>();
+  const [remainingTime, setRemainingTime] = useState<number>(-1);
+
+  const currentDate: Date = new Date();
+  const adjustedCurrent: Date = new Date(currentDate.getTime() + timeSync);
+
+  const calculateRemainingTime = () => {
+    const durationInMilliseconds = durationInSeconds * 1000;
+    const adjustedCurrentTime = adjustedCurrent.getTime();
+
+    return Math.floor(((referenceStartedTime + durationInMilliseconds) - adjustedCurrentTime) / 1000);
+  };
+
+  useEffect(() => {
+    if (remainingTime && durationInSeconds) {
+      if (durationInSeconds > 0 && timeRemainingInterval && referenceStartedTime) {
+        setRemainingTime(calculateRemainingTime());
+      }
+
+      clearInterval(timeRemainingInterval.current);
+      const remainingMillisecondsDiff = (
+        (referenceStartedTime + (durationInSeconds * 60000)) - adjustedCurrent.getTime()
+      ) % 1000;
+      timeRemainingInterval.current = setInterval(() => {
+        setRemainingTime((currentTime) => currentTime - 1);
+      }, remainingMillisecondsDiff === 0 ? 1000 : remainingMillisecondsDiff);
+    }
+
+    return () => {
+      clearInterval(timeRemainingInterval.current);
+    };
+  }, [remainingTime, durationInSeconds]);
+
+  let meetingTimeMessage: string = '';
+  let boldText: boolean = false;
+
+  if (remainingTime >= 0 && timeRemainingInterval) {
+    if (remainingTime > 0) {
+      const alertsInSeconds = REMAINING_TIME_ALERT_THRESHOLD_ARRAY.map((item) => item * 60);
+
+      if (alertsInSeconds.includes(remainingTime) && remainingTime !== lastAlertTime && displayAlerts) {
+        const timeInMinutes = remainingTime / 60;
+        const message = isBreakoutDuration
+          ? intlMessages.alertBreakoutEndsUnderMinutes
+          : intlMessages.alertMeetingEndsUnderMinutes;
+        const msg = { id: `${message.id}${timeInMinutes === 1 ? 'Singular' : 'Plural'}` };
+        const alertMessage = intl.formatMessage(msg, { 0: timeInMinutes });
+
+        lastAlertTime = remainingTime;
+        notify(alertMessage, 'info', 'rooms');
+      }
+
+      if (isBreakout) {
+        const breakoutMessage = intl.formatMessage(
+          intlMessages.breakoutTimeRemaining,
+          { 0: humanizeSeconds(remainingTime) },
+        );
+
+        return (
+          <span data-test="timeRemaining">
+            {breakoutMessage}
+          </span>
+        );
+      }
+
+      if (fromBreakoutPanel) boldText = true;
+      meetingTimeMessage = intl.formatMessage(fromBreakoutPanel || isBreakoutDuration
+        ? intlMessages.breakoutDuration
+        : intlMessages.meetingTimeRemaining, { 0: humanizeSeconds(remainingTime) });
+    } else {
+      clearInterval(timeRemainingInterval.current);
+      BreakoutService.setCapturedContentUploading();
```

Comment:
Make any consumption of data of meteor come from a service file, to make easy map and remove it in future